src/io/inputs/model-converter/convertorVisitor.cpp

```diff
@@ -389,32 +397,137 @@ std::any ConvertorVisitor::visitDual(ExprParser::DualContext* context)
         return node;
     }
 
-    throw NoConstraintWithThisName(constraint_id);
+    throw NoConstraintWithThisName(model_.id, constraint_id);
 }
 
-std::any ConvertorVisitor::visitReducedCost(ExprParser::ReducedCostContext* context)
+std::any ConvertorVisitor::handleReducedCost(ExprParser::ArgListContext* context)
 {
-    const auto& variables = model_.variables;
-    for (std::size_t index = 0; index < variables.size(); ++index)
+    std::vector<Node*> nodes;
+    try
     {
-        const auto& var = variables[index];
-        if (var.id == context->IDENTIFIER()->getText())
-        {
-            return static_cast<Node*>(
-              registry_.create<ReducedCostNode>(var.id,
-                                                index,
-                                                convertToTimeIndex(var.time_dependent,
-                                                                   var.scenario_dependent)));
-        }
+        nodes = std::any_cast<std::vector<Node*>>(context->accept(this));
+    }
+    catch (NoParameterOrVariableWithThisName) // to print accurate message
+    {
+        throw NoVariableWithThisName(model_.id, context->expr(0)->getText());
     }
+    if (nodes.size() != 1)
+    {
+        throw std::invalid_argument("reduced_cost operator expect only one variable id");
+    }
+
+    return static_cast<Node*>(
+      registry_.create<FunctionNode>(FunctionNodeType::reduced_cost, nodes.at(0)));
+}
+template<class T>
+std::any ConvertorVisitor::processPower(std::vector<T*> exprContexts,
+                                        const std::function<std::string()>& toStringTreeCallBack)
```

Comment:
Not sure if we really need a callback in this case. Since it accepts no arguments, it seems like a `std::string` would be enough.

Also, a `std::function` can be empty, `operator()` throws std::bad_function_call if *this does not store a callable function target, i.e. !*this == true.